src/components/ModalBackupWords.js

```diff
@@ -96,58 +100,85 @@ class ModalBackupWords extends React.Component {
+   * As the words were shown previosly, the user should be able to find any word given an index.
    */
   handleWordsSaved = () => {
-    const wordsValidation = _.shuffle(this.props.words.split(' ')).slice(0, WORDS_VALIDATION);
-    this.setState({ showValidation: true, wordsValidation });
-  }
+    const wordsArray = this.props.words.split(' ');
+    const wordsToValidate = _.shuffle(wordsArray).slice(0, WORDS_VALIDATION);
+    const validationSteps = wordsToValidate.map((word, index) => {
+      const backupIndex = wordsArray.indexOf(word);
+      let optionsStartIndex = backupIndex - 2;
+      let optionsEndIndex = backupIndex + 2;
+
+      // If index is 0 or 1, startIndex would be negative
+      // So we set start to 0 and end to 4
+      if (optionsStartIndex < 0) {
+        optionsStartIndex = 0;
+        optionsEndIndex = optionsStartIndex + 4;
+      }
+
+      // If index is 22 or 23, endIndex would be greater than the max
+      // So we set to the max and decrease the startIndex
+      const maxIndex = wordsArray.length - 1;
+      if (optionsEndIndex > maxIndex) {
+        optionsEndIndex = maxIndex;
+        optionsStartIndex = maxIndex - 4;
+      }
+
+      const options = _.shuffle(
+        wordsArray.slice(optionsStartIndex, optionsEndIndex + 1)
+      );
+
+      return {
+        backupIndex: backupIndex + 1,
+        correctWord: word,
+        options,
+        done: false,
+        last: index === wordsToValidate.length - 1,
+      };
+    });
+    this.setState({ showValidation: true, validationSteps });
+  };
 
   /**
    * Called when user finishes the backup. We validate the backup and shows a success message, in case of success
    */
-  handleValidateBackup = () => {
-    if (this.state.chosenWords.length !== WORDS_VALIDATION) {
-      this.setState({ errorMessage: t`Invalid number of words` });
+  handleValidationStepOption = (option, validationStep) => {
+    validationStep.chosenOption = option;
+    if (option !== validationStep.correctWord) {
+      this.setState({
+        validationStep,
+        errorMessage: (
+          <span>
+            {t`Wrong word. Please double check the words you saved.`}
+            <strong onClick={() => this.setState({ showValidation: false, errorMessage: '' })}>Click here to start the process again.</strong>
+          </span>
+        ),
+      });
       return;
     }
-    let validationArray = this.props.words.split(' ');
-    for (const word of this.state.chosenWords) {
-      const wordIdx = validationArray.indexOf(word);
-      if (wordIdx === -1) {
-        this.setState({ errorMessage: t`Invalid sequence of words` });
-        return;
-      }
-      validationArray = validationArray.slice(wordIdx);
+    if (validationStep.last) {
+      this.props.validationSuccess();
+      return;
     }
-    this.setState({ errorMessage: '' });
-    this.props.validationSuccess();
-  }
+    validationStep.done = true;
+    this.setState({ validationStep });
```

Comment:
Is this a new state? I don't see it in the definition of `this.state`

Reply:
Changing `this.state` directly is not a good practice, as it should be treated as immutable (more details here: https://reactjs.org/docs/react-component.html#state). We should use `this.setState` whenever is possible.

Reply:
I think he means explaining in the comments what `validationStep` means.

Reply:
Sorry. I added a comment about each `validationStep` property that will help to understand this object.

Reply:
Not only that, I don't see where `this.state.validationStep` is used, we have two different states `validationStep` and `validationSteps` and the first one is never used.

Reply:
Absolutely right. Put `validationStep` into the state did not make sense. Fixed.

Reply:
Done

Reply:
Done